src/rhsmlib/services/entitlement.py

```diff
@@ -0,0 +1,127 @@
+# version 2 (GPLv2). There is NO WARRANTY for this software, express or
+# implied, including the implied warranties of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+# along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+#
+# Red Hat trademarks are not licensed under GPLv2. No permission is
+# granted to use or replicate Red Hat trademarks that are incorporated
+# in this software or its documentation.
+import logging
+
+from subscription_manager import injection as inj
+from subscription_manager.i18n import ugettext as _
+from subscription_manager import managerlib, utils
+
+from rhsmlib.services import exceptions, products
+
+log = logging.getLogger(__name__)
+
+
+class EntitlementService(object):
+    def __init__(self, cp=None):
+        self.cp = cp
+        self.identity = inj.require(inj.IDENTITY)
+        self.product_dir = inj.require(inj.PROD_DIR)
+        self.entitlement_dir = inj.require(inj.ENT_DIR)
+
+    def get_status(self, on_date=None):
+        sorter = inj.require(inj.CERT_SORTER, on_date)
+        if self.identity.is_valid():
+            overall_status = sorter.get_system_status()
+            reasons = sorter.reasons.get_name_message_map()
+            valid = sorter.is_valid()
+            return {'status': overall_status, 'reasons': reasons, 'valid': valid}
+        else:
+            return {'status': 'Unknown', 'reasons': {}, 'valid': False}
+
+    def get_pools(self, pool_subsets=None, matches=None, pool_only=None, match_installed=None,
+            no_overlap=None, service_level=None, show_all=None, on_date=None, **kwargs):
+        # We accept a **kwargs argument so that the DBus object can pass whatever dictionary it receives
+        # via keyword expansion.
+        if kwargs:
+            raise exceptions.ValidationError(_("Unknown arguments: %s") % kwargs.keys())
+
+        # [] or None means look at all pools
+        if not pool_subsets:
+            pool_subsets = ['installed', 'consumed', 'available']
+
+        options = {
+            'pool_subsets': pool_subsets,
+            'matches': matches,
+            'pool_only': pool_only,
+            'match_installed': match_installed,
+            'no_overlap': no_overlap,
+            'service_level': service_level,
+            'show_all': show_all,
+            'on_date': on_date,
+        }
+        self.validate_options(options)
+        results = {}
+        if 'installed' in pool_subsets:
+            results['installed'] = products.InstalledProducts(self.cp).list(options['matches'])._as_dict()
+        if 'consumed' in pool_subsets:
+            results['consumed'] = self.get_consumed_product_pools(**options)
+        if 'available' in pool_subsets:
+            results['available'] = self.get_available_pools(**options)
+
+        return results
+
+    def get_consumed_product_pools(self, service_level=None, matches=None, **kwargs):
```

Comment:
`get_consumed_product_pools` will likely be called at some later point by `managercli`.

`**kwargs` is not used because it exists to absorb extra keyword arguments that aren't applicable to the method.  This allows us to say `**options` when calling these methods so we don't have to unpack and validate the options dictionary when calling `get_consumed_product_pools` and `get_available_pools`.